apps/src/code-studio/initApp/loadApp.js

```diff
@@ -1,17 +1,203 @@
-/* global dashboard, appOptions, addToHome */
+/* global dashboard, appOptions, addToHome script_path, CDOSounds, appOptions, trackEvent, Applab, Blockly */
 import $ from 'jquery';
 var renderAbusive = require('./renderAbusive');
 var userAgentParser = require('./userAgentParser');
 var progress = require('../progress');
 var clientState = require('../clientState');
 var color = require('../../color');
+import PlayZone from '@cdo/apps/code-studio/components/playzone';
+import React from 'react';
+import ReactDOM from 'react-dom';
+var timing = require('@cdo/apps/code-studio/initApp/timing');
+var chrome34Fix = require('@cdo/apps/code-studio/initApp/chrome34Fix');
+var project = require('@cdo/apps/code-studio/initApp/project');
+var createCallouts = require('@cdo/apps/code-studio/callouts');
+var reporting = require('@cdo/apps/code-studio/reporting');
+var Dialog = require('@cdo/apps/code-studio/dialog');
+var showVideoDialog = require('@cdo/apps/code-studio/videos').showVideoDialog;
+import { lockContainedLevelAnswers } from '@cdo/apps/code-studio/levels/codeStudioLevels';
+import queryString from 'query-string';
 
 import { activityCssClass, mergeActivityResult, LevelStatus } from '../activityUtils';
 
 // Max milliseconds to wait for last attempt data from the server
 var LAST_ATTEMPT_TIMEOUT = 5000;
 
-module.exports = function (callback) {
+// Legacy Blockly initialization that was moved here from _blockly.html.haml.
+// Modifies `appOptions` with some default values in `baseOptions`.
+// TODO(dave): Move blockly-specific setup function out of shared and back into dashboard.
+function setupApp(appOptions) {
+  if (!window.dashboard) {
+    throw new Error('Assume existence of window.dashboard');
+  }
+  timing.startTiming('Puzzle', script_path, '');
+
+  var lastSavedProgram;
+
+  if (appOptions.hasContainedLevels) {
+    if (appOptions.readonlyWorkspace) {
+      // Lock the contained levels if this is a teacher viewing student work:
+      lockContainedLevelAnswers();
+    }
+    // Always mark the workspace as readonly when we have contained levels:
+    appOptions.readonlyWorkspace = true;
+  }
+
+  // Sets up default options and initializes blockly
+  var baseOptions = {
+    containerId: 'codeApp',
+    Dialog: Dialog,
+    cdoSounds: CDOSounds,
+    position: {blockYCoordinateInterval: 25},
+    onInitialize: function () {
+      createCallouts(this.level.callouts || this.callouts);
+      if (userAgentParser.isChrome34()) {
+        chrome34Fix.fixup();
+      }
+      if (appOptions.level.projectTemplateLevelName || appOptions.app === 'applab' || appOptions.app === 'gamelab' || appOptions.app === 'weblab') {
```

Comment:
I know you didn't change this, but I strongly suspect this doesn't belong here.  Something to keep an eye on for future changes.